veps/sig-compute/tdx-sev-snp-proposal.md

```diff
@@ -0,0 +1,341 @@
+default way. All parameters are optional to configure a TDX VM, please refer to
+API Examples.
+
+### AMD Design
+
+AMD SEV-SNP maps to a new type in the LibVirt XML of type “sev-snp” with its
+own attributes and elements that are similar to the type “sev”.
+
+```xml
+   <domain>
+     ...
+     <launchSecurity type='sev-snp'>
+       <cbitpos>47</cbitpos>
+<policy>0x00030000</policy>
+       <reducedPhysBits>1</reducedPhysBits>
+	<guestVisibileWorkarounds>...</guestVisibleWorkaround>
+	<idBlock>...</idBlock>
+	<idAuth>...</idAuth>
+	<hostData>...</hostData>
+     </launchSecurity>
+     ...
+   </domain>
+```
+
+A new structure should be created for adding SEV-SNP on the `LaunchSecurity`
+field.
+```go
+type SEVSNP struct {
+    // 64-bit SEV-SNP Guest Policy
+    // +optional
+    Policy *string `json:"policy,omitempty"`
+    // 16-byte base64 encoded guest hypervisor-defined workarounds.
+    // +optional
+    GuestVisibleWorkarounds *string `json:"guestVisibleWorkarounds,omitempty"`
+    // 96-byte base64 encoded ID Block Structure.
+    // +optional
+    IDBlock *string `json:"idBlock,omitempty"`
+    // 4096-byte base64 encoded ID Auth Structure.
+    // +optional
+    IDAuth *string `json:"idAuth,omitempty"`
+    // 32-byte base64 encoded user-defined blob to provide to the guest.
+    // +optional
+    HostData *string `json:"hostData,omitempty"`
+    // Whether the guest is allowed to use VCEK for attestation reports. Set to false to disable VCEK usage.
+    // +optional
+    AuthorKey *bool `json:"authorKey,omitempty"`
+    // Whether idAuth contains VCEK field for attestation
+    // +optional
+    VCEK *bool `json:"vcek,omitempty"`
+    // Optional attribute to indicate whether the hashes of the kernel, and command line should be included in the measurement done by the firmware.
+    KernelHashes *bool `json:"kernelHashes,omitempty"`
+}
+```
+Similar to TDX, all elements for type=’sev-snp’, users should be able to
+provide additional settings to configure their Confidential VM. Some options
+should have a default value such as the policy if the user does not specify the
+VM should be using the QEMU default policy.  The node labller will detect
+SEV-SNP capabilities from the LibVirt domain capabilities then apply the label
+to the node, while the node selector renderer will be extended to include
+SEV-SNP scheduling.
+
+## API Examples
+The bellow yaml snippets provide examples of how to request TDX feature in the
+VMI spec.
+
+### API Examples for Intel TDX
+
+- TDX VM with defaults settings
+
+```yaml
+apiVersion: kubevirt.io/v1
+kind: VirtualMachineInstance
+metadata:
+  ...
+spec:
+  domain:
+    launchSecurity:
+      tdx: {}
+  ...
+```
+
+- TDX VM with user-defined parameters (no qgs)
+
+```yaml
+   spec:
+     domain:
+       launchSecurity:
+         tdx:
+           policy: 0x10000000
+           mrConfigId: xxx
+```
+
+- TDX VM with qgs settings, note libvirt will use `/var/run/tdx-qgs/qgs.socket`
```

Comment:
That seems outdated when qgs path was planned on the VMI api. KubeVirt should wire this transparently and either run with just TDX support or TDX+quote support.

Reply:
I will update this part once we decide whether QGS should be configured by KubeVirt

Reply:
I will remove this from the document.